extension/content/components/recipes/form/arguments/ShowHeartBeat.js

```diff
@@ -0,0 +1,135 @@
+  useRecipeDetailsDispatch,
+} from "devtools/contexts/recipeDetails";
+
+import InputField from "devtools/components/recipes/form/arguments/fields/InputField";
+import SelectField from "devtools/components/recipes/form/arguments/fields/SelectField";
+import ToggleField from "devtools/components/recipes/form/arguments/fields/ToggleField";
+
+export default function ShowHeart() {
+  return (
+    <FormGroup>
+      <Row>
+        <Col xs={12}>
+          <InputField label="Survey ID" name="surveyID" />
+          <InputField label="Message" name="message" />
+          <InputField label="Learn More Message" name="learnMoreMessage" />
+          <RepeatOptionsField />
+        </Col>
+        <Col xs={12}>
+          <InputField
+            label="Engagement Button Label"
+            name="engagementButtonLabel"
+          />
+          <InputField label="Thanks Message" name="thanksMessage" />
+          <InputField label="Post-Answer URL" name="postAnswerUrl" />
+          <ToggleField
+            label="Include Telemetry UUID?"
+            name="includeTelemetryUUID"
+          >
+            Include UUID in Post-Answer URL and Telemetry.
+          </ToggleField>
+        </Col>
+      </Row>
+    </FormGroup>
+  );
+}
+
+function RepeatOptionsField() {
+  const SHOW_PROMPT_OPTIONS = [
+    { label: "Show this prompt once.", value: "once" },
+    {
+      label: "Show users every X days.",
+      value: "xdays",
+    },
+    {
+      label:
+        "Show until the user interacts with the heartbeat prompt, then never again.",
+      value: "nag",
+    },
+  ];
+
+  const repeatsEveryChangeSideEffect = ({ data, value }) => {
+    if (value !== "xdays") {
+      /* eslint-disable no-unused-vars */
+      const {
+        repeatEvery: _omitRepeatOption,
+        action,
+        ...cleanedArguments
+      } = data.arguments;
+      /* eslint-enable no-unused-vars */
+      return { ...data, arguments: cleanedArguments };
+    }
+
+    return data;
+  };
+
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+
+  const handleChange = (name) => {
+    return (value) => {
+      dispatch({
+        type: ACTION_UPDATE_DATA,
+        data: {
+          ...data,
+          arguments: {
+            ...data.arguments,
+            [name]: parseNumericInput(value),
+          },
+        },
+      });
+    };
+  };
+
+  const parseNumericInput = (value) => {
+    if (!value) {
+      return "";
+    }
+
+    return parseInt(value, 10);
+  };
+
+  const repeatEveryOption = () => {
+    if (data.arguments.repeatOption !== "xdays") {
+      return null;
+    }
+
+    return (
+      <FormGroup>
+        <ControlLabel>Days before user is reprompted</ControlLabel>
+        <InputNumber
+          min={1}
+          postfix="days"
+          value={data.arguments.repeatEvery}
+          onChange={handleChange("repeatEvery")}
+        />
+        <HelpBlock>Required</HelpBlock>
+      </FormGroup>
+    );
+  };
```

Comment:
i think you can just split this out into its own function below:
```js
function RepeatEveryField() {
  const data = useRecipeDetailsData();

  if (data.arguments.repeatOption !== "xdays") {
    return null;
  }
  ...
```

Reply:
also i think the way you've set up this field it switches between controlled and uncontrolled and that's causing issues. you will need to make sure that `data.arguments.repeatEvery` has a numeric value or an empty string value before this field is rendered